[16.0] stock_release_channel_partner_public_holidays

nguyenminhchien opened the pull request:
Depend on: https://github.com/anothingguy/wms/pull/1

rousseldenis committed:
[16.0][ADD] stock_release_channel_process_end_date

sbejaoui committed:
[ADD] stock_release_channel_shipment_advice

This module is used to plan the  shipment advice for released pickings in a
stock release channel.

As this is the base module, it provides only a simple
planning mode. All released pickings are grouped together in a single advice.

fix label

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

[FIX] - button_show_shipment_advice: improve domain

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

[IMP] - _compute_can_plan_shipment: improve expression

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

[IMP] - _plan_shipment: improve expression

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

Apply suggestions from code review

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

rousseldenis committed:
[IMP] stock_release_channel_process_end_time: Add timezone for end time

As time should be timezoned to correctly compute the real end date, consider
the channel warehouse timezone or company one. If not defined, consider the timezone
as UTC.

rousseldenis committed:
[FIX] stock_release_channel_process_end_time: Don't take into account seconds and microseconds

sbejaoui committed:
[IMP] s_release_channel_shipment_advice: add dock to release channel

dock and warehouse are propgated from the release channel to shipment advice planner

sbejaoui committed:
[IMP] s_release_channel_shipment_advice: check dock WH

add data validation + form behavior improvement

sbejaoui committed:
[IMP] s_release_channel_shipment_advice: update description

rousseldenis committed:
[IMP] stock_release_channel_process_end_time: Check if end time is defined before assignation to scheduled_date

lmignon committed:
[FIX] stock_release_channel_process_end_time: releasable picking computation

The compute of releasable picking must take into account the process end date of the stock release channel and not the current datetime.
The way the ORM is working into Odoo prevent to makes query on model with criteria based on values from linked model (comparing column from model A to column of model B). To work arround this limitation, we use a specialized field with a search method. Field's search methods are considered as 'internal' search methods and therefore allows you to use a special 'inselect' operator. This kind operator is usefull since it allows to express a criteria as a plain SQL quey. Thanks to this approach we've an easy way to implement our needs with the Odoo's ORM.

sbejaoui committed:
[FIX] s_release_channel_shipment_advice: fix method param

sbejaoui committed:
[FIX] use picking type warehouse to filter pickings to plan

lmignon committed:
[IMP] s_release_channel_shipment_advice: speedup compute

sbejaoui committed:
[FIX] s_release_channel_shipment_advice: fix compute picking_to_plan

self in compute is composed of objects browse(NewId), setting the computed field
with the db id leads to odoo raises and error saying newId didn't get a value

rousseldenis committed:
[IMP] s_release_channel_shipment_advice: tree view default

Also filter on Today by default

sbejaoui committed:
[FIX] - assign channel end date to picking after release

sbejaoui committed:
[IMP] - improve process end time location in form view

sbejaoui committed:
[FIX] fix help message & add UTC as defauly value for process_end_time_tz

Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

jsanchezdeavila committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (73 of 73 strings)

Translation: wms-16.0/wms-16.0-stock_warehouse_flow
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_warehouse_flow/es/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: wms-16.0/wms-16.0-stock_release_channel_batch_mode_commercial_partner
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_release_channel_batch_mode_commercial_partner/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: wms-16.0/wms-16.0-stock_picking_type_shipping_policy
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_picking_type_shipping_policy/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: wms-16.0/wms-16.0-stock_release_channel_auto_release
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_release_channel_auto_release/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: wms-16.0/wms-16.0-stock_picking_completion_info
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_picking_completion_info/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 95.6% (44 of 46 strings)

Translation: wms-16.0/wms-16.0-stock_dynamic_routing
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_dynamic_routing/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (73 of 73 strings)

Translation: wms-16.0/wms-16.0-stock_warehouse_flow
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_warehouse_flow/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (46 of 46 strings)

Translation: wms-16.0/wms-16.0-stock_dynamic_routing
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_dynamic_routing/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: wms-16.0/wms-16.0-stock_release_channel_auto_release
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_release_channel_auto_release/es/

dzungtran89 commented:
@nguyenminhchien thanks for the PR, I got this error as running the tests
```  File "/home/dzungt/code/oca/wms/16.0/stock_release_channel_partner_public_holidays/tests/test_release_on_holiday.py", line 43, in test_assign_channel
    self.assertNotEqual(self.channel, self.picking.release_channel_id)
AssertionError: stock.release.channel(1,) == stock.release.channel(1,)
```

Could you please help to check?

nguyenminhchien commented:
> @nguyenminhchien thanks for the PR, I got this error as running the tests
> 
> ```
>     self.assertNotEqual(self.channel, self.picking.release_channel_id)
> AssertionError: stock.release.channel(1,) == stock.release.channel(1,)
> ```
> 
> Could you please help to check?

@dzungtran89 updated

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (75 of 75 strings)

Translation: wms-16.0/wms-16.0-stock_available_to_promise_release
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_available_to_promise_release/it/

nguyenminhchien committed:
[ADD] stock_release_channel_partner_public_holidays

anothingguy committed:
[IMP] add configuration to enable/disable set scheduled date

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-16.0/wms-16.0-stock_storage_type_putaway_abc
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_storage_type_putaway_abc/

anothingguy committed:
[ADD] stock_release_channel_shipment_lead_time

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: wms-16.0/wms-16.0-stock_storage_type_putaway_abc
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_storage_type_putaway_abc/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (110 of 110 strings)

Translation: wms-16.0/wms-16.0-stock_storage_type
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_storage_type/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (73 of 73 strings)

Translation: wms-16.0/wms-16.0-stock_warehouse_flow
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_warehouse_flow/it/

rousseldenis committed:
[IMP] stock_release_channel_process_end_time: Rename configuration parameter

nguyenminhchien committed:
[IMP] stock_storage_type: Don't rely on max_weight on product.packaging

OCA-git-bot committed:
Merge PR #760 into 16.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
s_release_channel_shipment_advice: update authorship and readme

jbaudoux committed:
s_release_channel_shipment_advice: fix form view

Move options in Options tab

OCA-git-bot committed:
Merge PR #761 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update stock_release_channel_shipment_advice.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #550 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update stock_release_channel_process_end_time.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #750 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update stock_release_channel_shipment_lead_time.pot

OCA-git-bot committed:
[BOT] post-merge updates
